### DISCUSSION TITLE
How can we set different size to each node throught nodeRelSize

### DISCUSSION THREAD

**Discussion by anujshaantruviz** [vasturiano/force-graph#365] [Show and tell]
I am creating a network graph and I have created node size based on the values inside .nodeCanvasObject() but when I am using onNodeHover to have hover interaction . The interaction isn't on whole node area because .nodeRelSize is having smaller area. We need to hover in the center of node to have hover interaction.

**Comment by vasturiano** [comment]
@anujshaantruviz please look into `nodePointerAreaPaint`. It exists to solve this exact problem.

Related to https://github.com/vasturiano/force-graph/issues/155 .

**Reply by anujshaantruviz** [reply]
Yeah I tried this property. But the issue in this is we have to hover over exact color (to match the code) to have interaction, right now I've circle node with pattern inside it now the pattern have colors which mean I've to hover over pattern to have interaction but I want interaction as soon as my pointer is inside the node circle area the hover interaction should work.

**Reply by anujshaantruviz** [reply]
for eg: in this link https://vasturiano.github.io/force-graph/example/custom-node-shape/
on the node "Text" If you hover between alphabets tooltip isn't coming we have to hover over exact alphabets

**Reply by vasturiano** [reply]
@anujshaantruviz in `nodePointerAreaPaint` you define specifically which area should be used for pointer interactions. If you want that to be a solid circle, then just draw a simple solid circle, without the cut-out pattern inside it.

**Reply by anujshaantruviz** [reply]
Thank you very much sir
